pkg/resources/statefulsets/log.go

```diff
@@ -162,6 +164,41 @@ func NewForLogDb(t *miniov2.Tenant, serviceName string) *appsv1.StatefulSet {
 	}
 	// if we have DB configurations to honor
 	if t.Spec.Log.Db != nil {
+
+		var initContainerSecurityContext corev1.SecurityContext
+
+		// If securityContext is present InitContainer still requires running with elevated privileges
+		// and user will have to provide a serviceAccount that allows this
+		if t.Spec.Log.Db.SecurityContext != nil {
+			var postgresRunAsUser int64
+			var postgresRunAsNonRoot bool
+			var postgresAllowPrivilegeEscalation bool
+
+			postgresRunAsUser = 0
+			postgresRunAsNonRoot = false
+			postgresAllowPrivilegeEscalation = true
+
```

Comment:
These values are set to default `0` aren't you supposed to set this with `t.Spec.Log.Db.SecurityContext.RunAsUser` ?

Reply:
Nop, this is the `initContainer` securityContext, it has to run as root to execute `chown`